Applications/FileIO/SWMM/SWMMInterface.cpp

```diff
@@ -0,0 +1,1248 @@
+        depth->reserve(max_depth.size());
+        std::copy(max_depth.cbegin(), max_depth.cend(), std::back_inserter(*depth));
+    }
+    else
+        ERR ("Size of max depth array does not fit number of elements. Skipping array.");
+
+    _mesh.reset(new MeshLib::Mesh(_base_name, nodes, elements, props));
+    return true;
+}
+
+bool SwmmInterface::matchSubcatchmentsWithOutlines(std::vector<GeoLib::Polyline*> const& lines, std::vector<std::string> const& names)
+{
+    std::size_t const n_lines (lines.size());
+    std::size_t const n_subcatchments (_subcatchments.size());
+
+    if (n_lines != n_subcatchments)
+    {
+        ERR ("Number of subcatchments does not match number of outlines.");
+        return false;
+    }
+    for (std::size_t i=0; i<n_lines; ++i)
+    {
+        bool found = false;
+        for (std::size_t j=0; j<n_subcatchments; ++j)
+        {
+            if (names[i] == _subcatchments[j].name)
+            {
+                _subcatchments[j].outline = lines[i];
+                found = true;
+                break;
+            }
+        }
+        if (found == false)
+        {
+            ERR ("No match in subcatcments for outline \"%s\".", names[i].c_str());
+            return false;
+        }
+    }
+    return true;
+}
+
+std::vector<std::string> SwmmInterface::getSubcatchmentNameMap() const
+{
+    std::vector<std::string> names;
+    names.reserve(_subcatchments.size());
+    for (auto sc : _subcatchments)
+        names.push_back(sc.name);
+    return names;
+}
+
+std::vector<std::string> SwmmInterface::getNames(SwmmObject obj_type) const
+{
+    switch (obj_type)
+    {
+    case SwmmObject::NODE:
+        return _id_nodename_map;
+    case SwmmObject::LINK:
+        return _id_linkname_map;
+    case SwmmObject::SUBCATCHMENT:
+        return getSubcatchmentNameMap();
+    case SwmmObject::SYSTEM:
+        std::vector<std::string> system_name { "System" };
+        return system_name;
+    }
+    ERR ("Object type has no name map");
+    std::vector<std::string> empty_vec;
+    return empty_vec;
+}
+
+std::string SwmmInterface::getName(SwmmObject obj_type, std::size_t idx) const
+{
+    switch (obj_type)
+    {
+    case SwmmObject::NODE:
+        if (idx < _id_nodename_map.size())
+            return _id_nodename_map[idx];
+    case SwmmObject::LINK:
+        if (idx < _id_linkname_map.size())
+            return _id_linkname_map[idx];
+    case SwmmObject::SUBCATCHMENT:
+        if (idx < _subcatchments.size())
+            return _subcatchments[idx].name;
+    case SwmmObject::SYSTEM:
+        if (idx == 0)
+            return std::string("System");
+    }
+    ERR ("Index out of bounds.");
+    return std::string("");
+}
+
+std::size_t SwmmInterface::getNObjects(SwmmObject obj_type) const
+{
+    std::string const outfile (_base_name + ".out");
+    std::size_t n_time_steps (std::numeric_limits<std::size_t>::max());
+    if (OpenSwmmOutFile(const_cast<char*>(outfile.c_str())) != 0)
```

Comment:
Instead of having `outfile` for one-time usage, you can simplify the whole story to:
```c++
if (OpenSwmmOutFile((_base_name + ".out").c_str()))
{
    ...
}
```